In [5]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, classification_report
from sklearn.linear_model import LogisticRegression

df = pd.read_csv("data/processed/stroke_processed.csv")

X = df.drop("stroke", axis=1)
y = df["stroke"]

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42, stratify=y)


In [4]:
y_pred_majority = [0] * len(y_test)

# Evaluate
print("Majority Class Baseline Performance:")
print(f"Accuracy: {accuracy_score(y_test, y_pred_majority):.4f}")
print(classification_report(y_test, y_pred_majority, zero_division=0))


Majority Class Baseline Performance:
Accuracy: 0.9511
              precision    recall  f1-score   support

           0       0.95      1.00      0.97       972
           1       0.00      0.00      0.00        50

    accuracy                           0.95      1022
   macro avg       0.48      0.50      0.49      1022
weighted avg       0.90      0.95      0.93      1022



The majority class classifier achieved an overall accuracy of 95.1%, but this result is misleading because it predicts no stroke for every case. While accuracy appears high due to class imbalance, the model completely fails to identify stroke cases (precision, recall, and F1-score of 0.00 for class 1). This baseline highlights the need for a more balanced and informative model for stroke prediction.

In [6]:
from sklearn.linear_model import LogisticRegression

log_reg = LogisticRegression(max_iter=1000)
log_reg.fit(X_train, y_train)

y_pred_log = log_reg.predict(X_test)

print("Logistic Regression Baseline Performance:")
print(classification_report(y_test, y_pred_log, zero_division=0))


Logistic Regression Baseline Performance:
              precision    recall  f1-score   support

           0       0.95      1.00      0.98       972
           1       1.00      0.02      0.04        50

    accuracy                           0.95      1022
   macro avg       0.98      0.51      0.51      1022
weighted avg       0.95      0.95      0.93      1022



The logistic regression baseline model achieved an overall accuracy of 95%, similar to the majority class classifier. However, while it slightly improved stroke detection recall of 0.02 for class 1, it still struggles to correctly identify positive cases due to severe class imbalance. This indicates that additional preprocessing steps, such as class balancing or feature scaling, will be necessary to improve model performance.